src/compiler/types/update-import-refs.ts

```diff
@@ -83,6 +91,21 @@ const updateImportReferenceFactory = (typeCounts: Map<string, number>, filePath:
           importResolvedFile = filePath;
         } else if (typeReference.location === 'import') {
           importResolvedFile = typeReference.path;
+
+          // We only care to resolve any _potential_ aliased
+          // modules if we're not already certain the path isn't an alias
+          if (!importResolvedFile.startsWith('.')) {
+            const { resolvedModule } = ts.resolveModuleName(
+              typeReference.path,
+              filePath,
+              tsOptions,
+              ts.createCompilerHost(tsOptions)
+            );
```

Comment:
Just FYI, this logic will currently transform the path aliases for the `src/components.d.ts` file we automatically generate at build time as well. This doesn't really cause any problems, but if we think that's confusing to users or just want to leave aliases in that file, I can update this logic a bit to ignore this block of code when writing to the `src` dir.

Aliases can exist there since it's not distributed code and the TS language server/any IDE would be able to handle the aliases in there.

Open to whatever the team thinks is best.